Facility manager section

gusferguson opened the pull request:
New section aimed at Facility Managers. Covers material suggested and fed back on https://trello.com/c/WziuUIoZ/66-new-chapter-facility-manager.

Needs complete review for content and accuracy, and proofing.

jburel commented:
travis failure with invalid URL

jburel commented:
```setups```  and ```set-ups```: need to be consistent.

jburel commented:
LDAP: maybe clarify that user will be in the "default" group so the user will need to be moved to the appropriate group

jburel commented:
```If permissions are reduced to Private when one member has made a projection of another members image in the group, the permissions change will fail.```  This is an example. Output of similar scripts will lead to the same error

jburel commented:
Admin tab of insight is buggy so I will probably remove it from the help

jburel commented:
```User Help for OMERO.dropbox are``` => is

jburel commented:
you have a mix of http and https link

jburel commented:
MCR (MATLAB Compiler Runtime) => specify a version?

jburel commented:
``System admins``=>``System administrators``

jburel commented:
The disc usage => disk

jburel commented:
```Details of how to embed thumbnails and links in web pages or Wikis are in the User Help Publishing with OMERO section.``` => Add a link to the section

jburel commented:
customisable =>customizable (if we go for US spelling)

jburel commented:
bespoke => very british word ;-)

gusferguson commented:
@jburel  - sorted URL issue - it was the " inherited from Word.

jburel commented:
@gusferguson: saw that. Thanks

gusferguson commented:
@jburel  - addressed most of the issues above except:

- you have a mix of http and https link - just what I have copied from web sites.
- MCR (MATLAB Compiler Runtime) => specify a version? - no - just means update overhead.
- customisable =>customizable (if we go for US spelling) - have always had proper English spelling in Help - US < 25% of visitors

gusferguson commented:
@hflynn @jburel - can we get this out with the 5.2.1 release?

hflynn commented:
No reason why not if we get it merged today. I'll have a look over it now. Have you got the pdf on staging?

gusferguson commented:
@hflynn - nothing on staging yet - was waiting for decision on whether to go for it - do you want me to add it all now? 
If so - should we include this in the zip of PDFs?

hflynn commented:
Hold off for just now, I'll read it through first.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<h2><a class="anchor" id="introduction"></a>Introduction</h2>
+
+<p>This section brings together descriptions of common workflows for Facility Managers and links to useful sections of User Help and Developer documentation.</p>
```
Developer documentation? I'd be inclined to say 'technical documentation' given you have a link to the sysadmin docs just below this but content you're covering here isn't 'developer' content as that is categorised in the sphinx docs.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<p>This is a frequently asked question, and the simplistic (if slightly unhelpful) answer is the best you can manage.</p>
+
+<p>Apart from basic CPU number, speed and physical disk size, here are a number of other issues to be taken into account, including the nature of the storage being used for the data, the connection type, and the speed of the network used to access the server.</p>
```
here -> there

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+<div class="line"><br /></div> <!-- end #line -->
+</div> <!-- end #line-block -->
+
+<h2><a class="anchor" id="using"></a>Using the OMERO server</h2>
```
I find the 'server' bit here misleading - this section discusses clients.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<h2><a class="anchor" id="permissions"></a>Permissions on OMERO</h2>
+
+<p>There are three levels of user permission in OMERO, Administrator, Group Owner and User. Each level of permission allows different actions to be performed on users own or other users data, and certain administrative tasks.</p>
```
All the 'users' in this section should be 'user's'

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<h2><a class="anchor" id="access"></a>Access control, authentication, user accounts</h2>
+
+<p>The recommended way to control access to the OMERO server is to configure integration with your institutes authentication system (most commonly LDAP).</p>
```
institute's

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+<div class="line"><br /></div> <!-- end #line -->
+</div> <!-- end #line-block -->
+
+<img alt="../images/facilityManagerUserWeb.jpg" class="align-center" src="images/facilityManagerUserWeb.jpg" style="width: 100%;" />
```
There's a fine line along the bottom of this screenshot which looks rather odd on staging

gusferguson replied:
@hflynn - can't see this on staging or image itself - can you try refresh with cmd key down and see if it disappears. If not can you screenshot and send it to me please.

hflynn replied:
Seems to be a Safari glitch, I can't reproduce in Chrome.

I just discovered your font doesn't look very bold in Chrome though - I can barely see the difference between bold and normal text :(

hflynn replied:
Firefox looks fine font-wise though and the line isn't there on that either, just Safari (which I'm using an out-of-date version of so don't worry)

gusferguson replied:
Looks fine in my Safari, but I will add a small margin to the bottom to see if that sorts it in yours.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<p class="first">Click on the <span class="bold">Users</span> tab.</p>
+
+<p>Click on the <span class="bold">Add new User</span> button.</p>
```
How is this 'removing a user from a group'? 
You're got the screenshots the wrong way round - edit user first in the groups tab section, and then the edit group screenshot in what should be the edit user tab description but has the wrong text

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+<div class="line"><br /></div> <!-- end #line -->
+</div> <!-- end #line-block -->
+
+<img alt="../images/facilityManagerRemoveUser.jpg" class="align-center" src="images/facilityManagerRemoveUser.jpg" style="width: 100%;" />
```
Repeated screenshot, doesn't belong here I don't think

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,791 @@
+
+<li>
+
+<p>Select <span class="bold">All Members</span> from the group display menu to view data owned by a user who is no longer a member of the group.</p>
```
These sections shouldn't be under a 'removing users…' subheading

hflynn commented:
Reviewed down to 'importing' section. Leaving off there for now as this obviously needs more work - will revisit post-release when we have more time.

gusferguson commented:
@hflynn - addressed all the issues above. The last issue was due to some text that had done astray in the Word-HTML transcription.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+<a href="#measure">Measurement and analysis</a><br />
+<a href="#disk">Disk usage</a><br />
+<a href="#email">Emailing users</a><br />
+<a href="#update">Updating</a><br />
```
I would use 'upgrading' rather than updating to be consistent with our technical docs.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+<div class="line"><br /></div> <!-- end #line -->
+</div> <!-- end #line-block -->
+
+<h2><a class="anchor" id="update"></a>Updating</h2>
```
Updating -> upgrading as above

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<h2><a class="anchor" id="using"></a>Using OMERO</h2>
+
+<p>Details on downloading the OMERO clients and accessing the OMERO server are in <a href="http://help.openmicroscopy.org/getting-started-5.html" target="_blank" class="external-url">Getting Started with OMERO</a> in User Help.</p>
```
This is now OMERO.insight in the menu, the different title here is a bit confusing

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>The recommended way to control access to the OMERO server is to configure integration with your institute's authentication system (most commonly LDAP).</p>
+
+<p>See the documentation for <a href="http://www.openmicroscopy.org/site/support/omero5.2/sysadmins/server-ldap.html" target="_blank" class="external-url">LDAP configuration</a> for details. Instructions for using Active Directory are at the bottom of this page.</p>
```
"…at the bottom of **this** page" implies this help guide, "…are also included" might be less confusing.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>Using LDAP authentication means new user accounts are created, and the user placed in the default group, the first time a user logs in, substantially reducing the administration load. Any change in the LDAP password by the user will then be reflected in the OMERO login.</p>
+
+<p>With LDAP authentication configured, an administrator can still create new accounts for non-LDAP users. In general, OMERO.web offers the best interface for Administrative tasks.</p>
```
'Administrative tasks' doesn't need a capital A

hflynn commented:
I would make all the docs links versionless (i.e. remove the 5.2 and just use /support/omero/ so that the redirects do the work) so you won't need to update the links when we move to 5.3.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>Select the new default group from the drop-down list.</p>
+
+<p>Click <span class="bold">Save</span> to save the new default group.</p>
```
Not a big fan of "click save to save" wording. 'Click to save the..." would read better, we don't need to tell people that a 'save' button saves something.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>Click to select the permissions level.</p>
+
+<p>Click <span class="bold">Save</span> to save the new group.</p>
```
Another 'click save to save' instance.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<ul>
+
+<li>view and move data owned by users no longer a member of a group</li>
```
Viewing data on the CLI seems a bit odd because elsewhere you use view to mean visualise. I'd be inclined to just say move, or use 'list' and move as in https://www.openmicroscopy.org/site/support/omero5.2/sysadmins/cli/fs.html

You might also want to reference https://www.openmicroscopy.org/site/support/omero5.2/users/cli/chgrp.html here as the CLI guides can be a little hard to find

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<li>view and move data owned by users no longer a member of a group</li>
+
+<li>change ownership of data.</li>
```
You don't need a full stop here

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>In the Import Location window select from the <span class="bold">Group</span> drop-down box.</p>
+
+<p>Select the user to from the <span class="bold">Import For</span> drop-down box.</p>
```
"…the user to from the…" -> some words missing here

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>OMERO.dropbox monitors designated folders on a storage system, detects newly uploaded files and runs a fully automatic import of those files to the OMERO server if possible. User Help for OMERO.dropbox is in the <a href="http://help.openmicroscopy.org/dropbox.html" target="_blank" class="external-url">Using OMERO.dropbox</a> page.</p>
+
+<p>OMERO.dropbox runs on the same machine as the OMERO server and is started automatically when the OMERO.server starts and it will run subject to prerequisites being met. Full details on configuring and using OMERO.dropbox are in the developer documentation <a href="https://www.openmicroscopy.org/site/support/omero5.2/sysadmins/dropbox.html" target="_blank" class="external-url">OMERO.dropbox</a> ...
```
"developer documentation" -> sysadmin documentation

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>More complex analysis can be done using the OMERO API that allows scripts written in Python, Matlab and other languages to use data from the OMERO server and save results back to the server.</p>
+
+<p>Further details on writing and installing scripts, the OMERO script repository and where to find other scripts are in <a href="http://www.openmicroscopy.org/site/community/scripts" target="_blank" class="external-url">Script Sharing</a> section  of the developer documentation.</p>
```
The script sharing page isn't in the developer documentation, I'd just say "…are on the script sharing page."

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<h2><a class="anchor" id="disk"></a>Disk usage</h2>
+
+<p>System administrators can view disk usage using the OMERO.web Admin tab. The disk usage reported reflects only data owned by the users or groups. Files that are uploaded to the server, or created in situ, and attached to Projects, Datasets or Images, are not included in the disk usage statistics.</p>
```
I'm confused as to the difference between data and files, does this mean it doesn't include attached files? Those are still data owned by users, right? Further clarity needed here I think and possibly a bug card filing as I'm wondering how much use this is if it excludes a lot of data.

gusferguson replied:
@hflynn - this is a well known issue and unlikely to be fixable for technical reasons. I will try to clarify the wording.

hflynn replied:
Ah ok, thanks. Would certainly be better to be more explicit about what is and isn't included I think.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<h2><a class="anchor" id="email"></a>Emailing users</h2>
+
+<p>The OMERO server can be configured so that administrators can use the OMERO.web Admin screen to email users through the OMERO system. Details on the configuration properties for mail are on <a href="https://www.openmicroscopy.org/site/support/omero5.2/sysadmins/config.html#mail" target="_blank" class="external-url">Configuration properties glossary</a> in the developer documentation.</p>
```
Again, sysadmin documentation not developer documentation

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+<div class="line"><br /></div> <!-- end #line -->
+</div> <!-- end #line-block -->
+
+<img alt="../images/sharingDataEmail.jpg" class="align-center" src="images/sharingDataEmail.jpg" style="width: 100%;" />
```
I see we still have the awful note at the bottom here, I need someone to show me where the code for this is so I can re-write it. Do we know what an inactive user is?

gusferguson replied:
Pass - Will or Ola will have to change it. 

An inactive user is what you do to one because you can't delete them.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>OMERO and Bio-Formats are developed and released independently, so new releases of Bio-Formats that address bugs or enable new data type handling are not held up by OMERO development and vice versa.</p>
+
+<p>OMERO point releases i.e. 5.2.x occur approximately quarterly, and breaking releases i.e. 5.x roughly bi-annually. Clients and servers are backwardly compatible for point releases, but not for breaking releases.</p>
```
I wouldn't suggest a timeline here, just that point releases occur regularly and only major releases are breaking and therefore not backwards compatible.

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<p>The full OMERO.web viewer can be embedded in pages as a customisable OMERO viewport. The level of interactivity can be set by adding buttons, links or scalebars etc. as desired. Details on the viewport can be seen in <a href="https://www.openmicroscopy.org/site/support/omero5.2/developers/Web/ViewPort.html#embedding-web-viewport" target="_blank" class="external-url">Customizing the content of  ...
+
+<p>OMERO is widely used as the basis for public data repositories. OMERO.gallery provides a basic gallery front end based on the Group-Project-Dataset-Image, and is a good starting point to develop a bespoke front end. Full documentation on how to set up OMERO as a public repository and details of customisation options can be found in the OMERO Repository documentation.</p>
```
repository documentation at the end should link to https://www.openmicroscopy.org/site/support/omero/sysadmins/index.html#optimizing-omero-as-a-data-repository

hflynn commented on facility-manager.html:
```diff
@@ -0,0 +1,806 @@
+
+<h2><a class="anchor" id="training"></a>Getting users started</h2>
+
+<p>Experience has indicated that an effective way of getting users started with OMERO is by including a short introduction to OMERO when users are given their introduction to the microscopy facilities. As this is when the users are establishing their data workflows, and establishing file storage systems, ensuring log-ins working and the user is in the correct groups and knows which group to save  ...
```
the second sentence of this paragraph could do with being re-worded and split up if needed

hflynn commented:
As one final point, could you sort the formatting re:line lengths, as the continuous lines make this very hard to review in the GH interface, please?

Otherwise, definitely getting there but would probably benefit from a fresh pair of eyes in the new year.

gusferguson commented:
@hflynn - thanks for all the feedback - I will sort it out today. 

Regarding the continuous lines - it is bad coding practice to put hard line breaks into HTML code and I am not sure it wouldn't cause problems with Jekyll. The fact that Git does not wrap HTML code in the diff viewer is a source of complaint by many people and there have been feature requests for it which they have chosen to ignore. Sorry!

hflynn commented:
Ah, I suppose I only use Jekyll for markdown files in the blog, I haven't tried building html files with it.

gusferguson commented:
@hflynn - I have put the updated PDF up on downloads staging but not added it to the Zip or done anything else yet.

hflynn commented:
New commits look good

sbesson commented on facility-manager.html:
```diff
@@ -0,0 +1,808 @@
+
+<p>OMERO and Bio-Formats are developed and released independently, so new releases of Bio-Formats that address bugs or enable new data type handling are not held up by OMERO development and vice versa.</p>
+
+<p>OMERO point releases, i.e. 5.2.x, occur regularly, and clients and servers are backwardly compatible for these. This is not the case for breaking releases, i.e. 5.x.</p>
```
To be more specific a breaking release should always end up with 0 as the patch number `i.e. 5.x.0`.

sbesson commented:
One minor nitpick. Otherwise the new chapter generally reads well. My major question is about the visibility of this new chapter. The new facility managers chapter is contained under `More Workflow User Guides` container which is folded by default when opening the help website. Having a separate container with only one chapter might be overkill for now but would that be something to consider in the future?

On the [HTML guidelines front](https://github.com/openmicroscopy/ome-help/pull/138#issuecomment-166231396), do we have a authoritative reference for the conventions used in the Help website? Some [online guidelines](http://www.w3schools.com/html/html5_syntax.asp) suggest to cap the line width to increase readability. This is definitely not a blocker for this PR but the topic is worth discussing next year and updating the readme accordingly.

hflynn commented:
Re: menu placing - we will have to think carefully about where we promote this new guide I think. As well as adding the link to docs index pages (the main index and the sysadmin pages), we'll want to add it to the 'Running OMERO for your institution' section on https://www.openmicroscopy.org/site/products/omero, https://www.openmicroscopy.org/site/support/new-users and maybe even the home page.

gusferguson commented:
@hflynn @sbesson 

I agree with the placing in the menu. The only suggestion I have is to put it at the top of the "More" section, where it would at least be visible by default.

Let me know if you want to try that.

Regarding the long lines - the problem with putting hard line breaks in are: 1. I am not sure Jekyll will handle them, 2. it means that if your text editor doesn't wrap to that, but smaller, then you end up with an unreadable mess. We could put a hard wrap at the end of every sentence, but I am not sure that would help a lot.

We can look into it further after the break.

hflynn commented:
I think the top item of the 'More' menu (i.e. above Troubleshooting) may be the best place for it, yes.

hflynn commented:
Menu looks better and last commit is fine too.

gusferguson commented:
Do you want to leave it until the New Year to merge? Or go the full wellie now?

hflynn commented:
Let's leave it for the NY and let someone else give it one final once over. Pdf still need checking too, I haven't looked at that yet.

jburel commented:
@hflynn, @gusferguson: I will try to go over it during the break.

ximenesuk commented:
@hflynn is there a way to access the pdf through staging? The link in the staging webpage is to where the final doc will be.

hflynn commented:
Yes, it's at http://downloads.openmicroscopy.org/help-staging/pdfs/facility-manager.pdf (you'll need vpn externally to access it)

ximenesuk commented on facility-manager.html:
```diff
@@ -0,0 +1,808 @@
+
+<p>Group structures and permissions are completely flexible in OMERO, so it is worthwhile taking some time to draw up a policy for the structure and naming of groups on your OMERO server that will meet the requirements of your facility.</p>
+
+<p>In Biology institutes, groups generally reflect the laboratory structure of the institute, with groups for each lab, groups containing members from different labs for cross-lab collaborative projects, and groups for cross-facility and inter-institution collaboration.</p>
```
Biology -> biology

gusferguson replied:
or would "Life Sciences institutes" be better?

hflynn replied:
Life Sciences institutes works for me

ximenesuk commented:
Other than that one minor point above this reads well and the layout looks fine in the PDF and web (using Chrome).

gusferguson commented:
@ximenesuk - Thanks Colin.

hflynn commented:
:+1:
